### PR TITLE
Observability: implement proxy tracing

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/envoyproxy/gateway/issues/699

TODO: add e2e test

### CODE REVIEW

File: internal/xds/translator/accesslog.go [hunk 1]
```diff
@@ -127,7 +126,7 @@ func buildXdsAccessLog(al *ir.AccessLog, forListener bool) []*accesslog.AccessLo
 				GrpcService: &cfgcore.GrpcService{
 					TargetSpecifier: &cfgcore.GrpcService_EnvoyGrpc_{
 						EnvoyGrpc: &cfgcore.GrpcService_EnvoyGrpc{
-							ClusterName: buildClusterNameForOpenTelemetry(otel),
```
Comment (arkodg) [inline internal/xds/translator/accesslog.go:130]: why is this deleted ? dont we need a unique naming if connecting to different otel collectors ?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by AliceProxy — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
